Etg.Yams.Test/Etg.Yams.Test/Storage/LocalDeploymentRepository.cs

```diff
@@ -0,0 +1,109 @@
+            return Task.FromResult(new DeploymentConfig(data));
+        }
+
+        public Task PublishDeploymentConfig(DeploymentConfig deploymentConfig)
+        {
+            File.WriteAllText(_deploymentConfigPath, deploymentConfig.RawData());
+            return Task.CompletedTask;
+        }
+
+        public Task UploadApplicationBinaries(AppIdentity appIdentity, string localPath, ConflictResolutionMode conflictResolutionMode)
+        {
+            if (FileUtils.DirectoryDoesntExistOrEmpty(localPath))
+            {
+                throw new BinariesNotFoundException(
+                    $"Binaries were not be uploaded because they were not found at the given path {localPath}");
+            }
+
+            string destPath = GetBinariesPath(appIdentity);
+            bool binariesExist = FileUtils.DirectoryDoesntExistOrEmpty(destPath);
+            if (binariesExist)
+            {
+                if (conflictResolutionMode == ConflictResolutionMode.DoNothingIfBinariesExist)
+                {
+                    return Task.CompletedTask;
+                }
+                if (conflictResolutionMode == ConflictResolutionMode.FailIfBinariesExist)
+                {
+                    throw new DuplicateBinariesException();
+                }
+            }
+            return FileUtils.CopyDir(_path, localPath, true);
+        }
+
+        private string GetBinariesPath(AppIdentity appIdentity)
+        {
+            return Path.Combine(_path, GetDeploymentRelativePath(appIdentity));
+        }
+
+        private static string GetDeploymentRelativePath(AppIdentity appIdentity)
+        {
+            return Path.Combine(appIdentity.Id, appIdentity.Version.ToString());
+        }
+
+        public Task DeleteApplicationBinaries(AppIdentity appIdentity)
+        {
+            string path = GetBinariesPath(appIdentity);
+            if (FileUtils.DirectoryDoesntExistOrEmpty(path))
+            {
+                throw new BinariesNotFoundException(
+                    $"Cannot delete binaries for application {appIdentity} because they were not found");
+            }
+            Directory.Delete(path, true);
+            return Task.CompletedTask;
+        }
+
+        public Task<bool> HasApplicationBinaries(AppIdentity appIdentity)
+        {
+            string path = GetBinariesPath(appIdentity);
+            return Task.FromResult(!FileUtils.DirectoryDoesntExistOrEmpty(path));
+        }
+
+        public async Task DownloadApplicationBinaries(AppIdentity appIdentity, string localPath, ConflictResolutionMode conflictResolutionMode)
+        {
+            bool exists = !FileUtils.DirectoryDoesntExistOrEmpty(localPath);
+            if (exists)
+            {
+                if (conflictResolutionMode == ConflictResolutionMode.DoNothingIfBinariesExist)
+                {
+                    return;
+                }
+                if (conflictResolutionMode == ConflictResolutionMode.FailIfBinariesExist)
```

Comment:
I don't understood why we would have conflicts... By having this, the storage will only have the latest version deployed there. So if I want to rollback a deployment for whatever reason, I should push the older bit again? If this is correct, I think the target storage should keep all the previous versions(or at least a configurable number of it - lets say, keep the last 2 versions) so the user can quickly select which one to rollback to.